### DISCUSSION TITLE
Suggestion: Add a watermark to code blocks indicating programming language

### DISCUSSION THREAD

**Discussion by jasper-zanjani** [squidfunk/mkdocs-material#2888] [Ideas]
I've been tinkering with admonitions and tabs in my notes repo for a while now, trying to find a visually appealing way to present the same coding task in various programming languages. I think what would actually be a great solution to this is a simple watermark icon, perhaps in the bottom right of the code block.  Obviously this could easily use the material programming language icons like `:material-language-python:` etc. Would this also be a pymdownx thing?

**Comment by facelessuser** [comment]
Material uses pymdownx which uses Pygments. Pygments doesn't really inject such things. I am working on exposing more Pygments options, one of which allows you to add a filename/title to a code block. This won't give you a watermark, but could give you a header with the code name if you so choose. Of course, this new filename feature would require some new CSS to look good, but I'm playing with it. We'll see how things go 🙂.
